src/mlpack/methods/kde/kde_rules_impl.hpp

```diff
@@ -240,6 +428,27 @@ EvaluateKernel(const arma::vec& query, const arma::vec& reference) const
   return kernel.Evaluate(metric.Evaluate(query, reference));
 }
 
+template<typename MetricType, typename KernelType, typename TreeType>
+inline force_inline size_t KDERules<MetricType, KernelType, TreeType>::
+CalculateDepth(const TreeType& node) const
+{
+  size_t depth = 0;
+  const TreeType* currentNode = &node;
+  while (currentNode != NULL)
+  {
+    ++depth;
+    currentNode = currentNode->Parent();
+  }
+  return depth;
+}
+
+template<typename MetricType, typename KernelType, typename TreeType>
+inline force_inline double KDERules<MetricType, KernelType, TreeType>::
+CalculateAlpha(const size_t depth) const
+{
+  return mcBeta / std::pow(2, depth + 1);
```

Comment:
Ahh... I realized something frustrating.  For kd-trees, this code works properly because each node has two children, and we "split" the probability equally among them.  But cover trees and other tree types may have arbitrary numbers of children.  Suppose we have a cover tree node with three children... then, each child should receive 1/3 of the probability of the parent, not 1/2.

I think the way to fix this may be a more complex algorithm that stores the computed `alpha` in the statistic, and when recursion is performed, the new `alpha` is computed based on the parent's number of children and the parent's `alpha` value.  Let me know if I can clarify what I mean.

Reply:
This could also cause the sum of alphas to be too large across trees.  (That will be true for cover trees, most likely, but it shouldn't be true for kd-trees.)

Reply:
This should be solved with 253332afaa68faf973c80a2742f0afc4432e65ea. Thanks for pointing this out :)